modules/core/shared/src/main/scala/lucuma/core/model/sequence/ExecutionSequence.scala

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.core.model.sequence
+
+import cats.Eq
+import cats.syntax.all._
+import monocle.Focus
+import monocle.Lens
+import monocle.Prism
+import monocle.macros.GenPrism
+
+sealed trait ExecutionSequence {
+  val nextAtom: Atom
+  val possibleFuture: List[Atom]
+}
```

Comment:
This pattern seems cleaner that what I did, where everything is parameterized by static and dynamic type all the way to the top.  I think what stopped me from going this way was that given an `ExecutionSequence` as defined above you have to match on the instances to do anything with it.  Otherwise the `Atom`s could in theory be from a mix of instruments.  This does look a lot cleaner though, so I'm very much okay with it if that doesn't seem like a problem to you.

Reply:
I struggled with this. I started by type-parametrizing, but that's unsafe to check at runtime, so I went with pattern matching, and then we can decide what we are dealing with at any point in the hierarchy.

I agree that the trait doesn't prevent mixed instances in the properties, but the types are fixed in the concrete classes, so mixed instances are not buildable. If it makes things clearer though, I think we could remove the general properties from the trait and just leave them in the concrete classes.

Reply:
> I went with pattern matching, and then we can decide what we are dealing with at any point in the hierarchy

I like it. Let's go with it.